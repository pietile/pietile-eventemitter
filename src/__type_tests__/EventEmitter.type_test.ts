import { EventEmitter } from '..';

interface Events {
  foo: () => void;
  bar: (a: number) => void;
}

const emitter = new EventEmitter<Events>();

// @dts-jest:fail:snap wrong emit argument
emitter.emit('foo', 1);

// @dts-jest:fail:snap wrong emit argument type
emitter.emit('bar', '1');

// @dts-jest:fail:snap wrong emit arguments count
emitter.emit('bar', 1, 2);

// @dts-jest:fail:snap wrong handler argument
emitter.on('foo', (a: number) => {});

// @dts-jest:fail:snap wrong handler argument type
emitter.on('bar', (a: string) => {});

// @dts-jest:fail:snap wrong handler arguments count
emitter.on('bar', (a: number, b: string) => {});
