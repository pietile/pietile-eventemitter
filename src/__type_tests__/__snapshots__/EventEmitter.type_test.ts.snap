// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wrong emit argument (type) should match snapshot 1`] = `"Expected 1 arguments, but got 2."`;

exports[`wrong emit argument type (type) should match snapshot 1`] = `"Argument of type '\\"1\\"' is not assignable to parameter of type 'number'."`;

exports[`wrong emit arguments count (type) should match snapshot 1`] = `"Expected 2 arguments, but got 3."`;

exports[`wrong handler argument (type) should match snapshot 1`] = `"Argument of type '(a: number) => void' is not assignable to parameter of type '() => void'."`;

exports[`wrong handler argument type (type) should match snapshot 1`] = `
"Argument of type '(a: string) => void' is not assignable to parameter of type '(a: number) => void'.
  Types of parameters 'a' and 'a' are incompatible.
    Type 'number' is not assignable to type 'string'."
`;

exports[`wrong handler arguments count (type) should match snapshot 1`] = `"Argument of type '(a: number, b: string) => void' is not assignable to parameter of type '(a: number) => void'."`;
